ios/browser/ui/webui/brave_webui_data_source.mm

```diff
@@ -0,0 +1,304 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/ios/browser/ui/webui/brave_webui_data_source.h"
+
+#include "base/functional/bind.h"
+#include "base/memory/raw_ptr.h"
+#include "base/memory/ref_counted_memory.h"
+#include "base/strings/string_util.h"
+#include "base/strings/utf_string_conversions.h"
+#include "ios/web/public/web_client.h"
+#include "services/network/public/mojom/content_security_policy.mojom.h"
+#include "ui/base/webui/jstemplate_builder.h"
+#include "ui/base/webui/resource_path.h"
+#include "ui/base/webui/web_ui_util.h"
+
+namespace {
+const char kChromeUIScheme[] = "chrome";
+const char kChromeUIUntrustedScheme[] = "chrome-untrusted";
```

Comment:
nit: I see this constant is being redefined a few times in this PR. I also did a quick search and don't see us using constants typically currently when defining URLs either and instead just putting the entire string of the WebUI URL in. I think it's probably better to either reuse the [content::kChromeUiUntrustedScheme](https://source.chromium.org/chromium/chromium/src/+/main:content/public/common/url_constants.h;l=23?q=kChromeUIUntrustedScheme%20%3D%20%22chrome-untrusted%22&ss=chromium) when we want just the scheme and to hard code it as a URL whenever we're defining urls.

Reply:
We're unable to access `content/` on iOS unfortunately. However Chromium defines `kChromeUIScheme` in [web_ui_url_constants.h](https://source.chromium.org/chromium/chromium/src/+/main:ios/components/webui/web_ui_url_constants.h) / [web_ui_url_constants.cc](https://source.chromium.org/chromium/chromium/src/+/main:ios/components/webui/web_ui_url_constants.cc). I've added a chromium src override to add `kChromeUIUntrustedScheme` to this file (they already have`kChromeUIScheme`)  and use those throughout our code instead of defining it in a few spots, and additionally updated a few spots where we were building a string to use the constants too.

https://github.com/brave/brave-core/pull/24328/commits/01d2f0d026da7c6e6fe53085eb8c889499fd35d4

Reply:
LGTM, thanks for addressing.